### ISSUE TITLE
Training a new language from scratch: No comprehensible output after 19 epochs with slow convergence

### ISSUE THREAD

**Issue by CHIHHSIANGLI** [SWivid/F5-TTS#795]
### Checks

- [x] This template is only for usage issues encountered.
- [x] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [x] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [x] I confirm that I am using English to submit this report in order to facilitate communication.

### Environment Details

F5-Small model

config:

hydra:
  run:
    dir: ckpts/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}/${now:%Y-%m-%d}/${now:%H-%M-%S}
    
datasets:
  name: cantonese
  batch_size_per_gpu: 10  # 8 GPUs, 8 * 38400 = 307200
  batch_size_type: sample  # "frame" or "sample"
  max_samples: 0  # max sequences per batch if use frame-wise batch_size. we set 32 for small models, 64 for base models
  num_workers: 16

optim:
  epochs: 20000
  learning_rate: 7.5e-5
  num_warmup_updates: 20000  # warmup steps
  grad_accumulation_steps: 4  # note: updates = steps / grad_accumulation_steps
  max_grad_norm: 1.0  # gradient clipping
  bnb_optimizer: False  # use bnb 8bit AdamW optimizer or not

model:
  name: F5TTS_Small
  tokenizer: custom
  tokenizer_path: /cantonese_vocab/vocab.txt  # if tokenizer = 'custom', define the path to the tokenizer you want to use (should be vocab.txt)
  arch:
    dim: 768
    depth: 18
    heads: 12
    ff_mult: 2
    text_dim: 512
    conv_layers: 4
    checkpoint_activations: False  # recompute activations and save memory for extra compute
  mel_spec:
    target_sample_rate: 24000
    n_mel_channels: 100
    hop_length: 256
    win_length: 1024
    n_fft: 1024
    mel_spec_type: vocos  # 'vocos' or 'bigvgan'
  vocoder:
    is_local: False  # use local offline ckpt or not
    local_path: None  # local vocoder path

ckpts:
  logger: wandb  # wandb | tensorboard | None
  save_per_updates: 50000  # save checkpoint per steps
  last_per_steps: 5000  # save last checkpoint per steps
  save_dir: ckpts/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}

### Steps to Reproduce

I'm training Cantonese from scratch with approximately 600+ hours of data. The training is being conducted on 2 V100 GPUs with a batch size type of "sample" and gradient accumulation steps of 4. After 19 epochs, the output is still incomprehensible, though I've verified that the frontend processing and corresponding vocab IDs are correct. The loss is converging normally.
I'm wondering if this is simply a matter of insufficient training time, or if there might be other underlying issues that need to be addressed?
Technical details:

![Image](https://github.com/user-attachments/assets/b09ce4f0-ca98-4639-98a1-a8456d596552)

**Comment by SWivid** [comment]
Hi @CHIHHSIANGLI , there is discussion #548 ~

**Comment by CHIHHSIANGLI** [comment]
@SWivid Can I ask about your training epochs, and what was your final loss value? I've read your discussion, but since I don't know your total sample size, I don't have a good sense of the number of steps. So I'd like to know after how many epochs did you achieve a good model performance?

**Comment by SWivid** [comment]
actually you could find these from e.g.
https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/train/datasets/prepare_libritts.py
and a simple tool
https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/scripts/count_max_epoch.py
